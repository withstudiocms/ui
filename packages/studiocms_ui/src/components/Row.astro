---
import type { HTMLAttributes } from 'astro/types';

interface Props extends HTMLAttributes<'div'> {
	alignCenter?: boolean;
	gapSize?: 'sm' | 'md' | 'lg';
};

const { alignCenter, gapSize = 'md', ...props } = Astro.props;
---

<div class="sui-row" class:list={[alignCenter && "align", gapSize]} {...props}>
  <slot />
</div>
<style>
	.sui-row {
		display: flex;
		flex-direction: row;
		position: relative;
		flex-wrap: wrap;
	}

	.sui-row.align {
		align-items: center;
	}

	.sui-row.sm {
		gap: .5rem;
	}

	.sui-row.md {
		gap: 1rem;
	}

	.sui-row.lg {
		gap: 2rem;
	}
</style>
