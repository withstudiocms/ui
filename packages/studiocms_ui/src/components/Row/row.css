.sui-row {
  display: flex;
  flex-direction: row;
  position: relative;
  flex-wrap: wrap;
}

.sui-row.align {
  align-items: center;
}

.sui-row.sm {
  gap: .5rem;
}

.sui-row.md {
  gap: 1rem;
}

.sui-row.lg {
  gap: 2rem;
}
