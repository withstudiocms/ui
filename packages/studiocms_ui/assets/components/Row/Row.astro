---
import type { HTMLAttributes } from 'astro/types';
import './row.css';

/**
 * The props for the row component.
 */
interface Props extends HTMLAttributes<'div'> {
	/**
	 * Whether the row should be aligned to the center. Defaults to `false`.
	 */
	alignCenter?: boolean;
	/**
	 * The size of the gap between the children. Defaults to `md`.
	 */
	gapSize?: 'sm' | 'md' | 'lg';
}

const { alignCenter, gapSize = 'md', ...props } = Astro.props;
---

<div class="sui-row" class:list={[alignCenter && "align", gapSize]} {...props}>
  <slot />
</div>
