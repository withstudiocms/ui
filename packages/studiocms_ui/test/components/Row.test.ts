import { describe, expect } from 'vitest';
import Row from '../../src/components/Row/Row.astro';
import { test } from '../fixtures/vitest/AstroContainer';

const mockContent = '<span>Small Row</span> <span>Small Row</span> <span>Small Row</span>';

describe('Row Component', () => {
	test('renders Row component correctly', async ({ renderComponent }) => {
		const result = await renderComponent(Row, 'Row', { slots: { default: mockContent } });
		expect(result).toMatchSnapshot();
	});

	test.for([{ alignCenter: true }, { gap: 'sm' }, { gap: 'md' }, { gap: 'lg' }])(
		'renders Row with props: "%s"',
		async (props, { renderComponent }) => {
			const result = await renderComponent(Row, 'Row', { props, slots: { default: mockContent } });
			expect(result).toMatchSnapshot();
		}
	);
});
