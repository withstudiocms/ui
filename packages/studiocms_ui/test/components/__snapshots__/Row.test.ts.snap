// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Row Component > renders Row component correctly 1`] = `"<div class="sui-row md"> <span>Small Row</span> <span>Small Row</span> <span>Small Row</span> </div>"`;

exports[`Row Component > renders Row with props: "{ alignCenter: true }" 1`] = `"<div class="sui-row align md"> <span>Small Row</span> <span>Small Row</span> <span>Small Row</span> </div>"`;

exports[`Row Component > renders Row with props: "{ gap: 'lg' }" 1`] = `"<div class="sui-row md" gap="lg"> <span>Small Row</span> <span>Small Row</span> <span>Small Row</span> </div>"`;

exports[`Row Component > renders Row with props: "{ gap: 'md' }" 1`] = `"<div class="sui-row md" gap="md"> <span>Small Row</span> <span>Small Row</span> <span>Small Row</span> </div>"`;

exports[`Row Component > renders Row with props: "{ gap: 'sm' }" 1`] = `"<div class="sui-row md" gap="sm"> <span>Small Row</span> <span>Small Row</span> <span>Small Row</span> </div>"`;
